### ISSUE TITLE
JDK11 JIT Server : MathLoadTest_all_special_12_FAILED net.adoptopenjdk.test.bigdecimal.TestSuite034

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#10331]
Failure link
------------

From an internal build `Test_openjdk11_j9_special.system_x86-64_linux_jit_Nightly_mathLoadTest/71`:
```
02:17:06  openjdk version "11.0.8-internal" 2020-07-14
02:17:06  OpenJDK Runtime Environment (build 11.0.8-internal+0-adhoc.jenkins.BuildJDK11x86-64linuxjitNightly)
02:17:06  Eclipse OpenJ9 VM (build master-1c194db6e72, JRE 11 Linux amd64-64-Bit Compressed References 20200803_145 (JIT enabled, AOT enabled)
02:17:06  OpenJ9   - 1c194db6e72
02:17:06  OMR      - 7b76a160bc7
02:17:06  JCL      - 94b9d6d2c6f based on jdk-11.0.8+10)
```

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------

```
MLT 20:51:04.453 - Completed 64.9%. Number of tests started=3405 (+440)
MLT 20:51:12.207 - First failure detected by thread: load-1. Not creating dumps as no dump generation is requested for this load test
MLT 20:51:12.211 - Test failed
MLT   Failure num.  = 1
MLT   Test number   = 34
MLT   Test details  = 'JUnit[net.adoptopenjdk.test.bigdecimal.TestSuite034]'
MLT   Suite number  = 0
MLT   Thread number = 1
MLT >>> Captured test output >>>
MLT Test failed:
MLT java.lang.InternalError: java.lang.reflect.InvocationTargetException
MLT 	at java.base/java.lang.Class.newInternalError(Class.java:3854)
MLT 	at java.base/java.lang.Class.lookupCachedMethod(Class.java:3874)
MLT 	at java.base/java.lang.Class.getMethodHelper(Class.java:1386)
MLT 	at java.base/java.lang.Class.getMethod(Class.java:1338)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.getAnnotations(JUnit38ClassRunner.java:131)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.makeDescription(JUnit38ClassRunner.java:101)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.makeDescription(JUnit38ClassRunner.java:109)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.getDescription(JUnit38ClassRunner.java:95)
MLT 	at org.junit.runners.Suite.describeChild(Suite.java:123)
MLT 	at org.junit.runners.Suite.describeChild(Suite.java:27)
MLT 	at org.junit.runners.ParentRunner.getDescription(ParentRunner.java:352)
MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
MLT 	at net.adoptopenjdk.loadTest.adaptors.JUnitAdaptor.executeTest(JUnitAdaptor.java:130)
MLT 	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:182)
MLT 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
MLT 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
MLT 	at java.base/java.lang.Thread.run(Thread.java:836)
MLT Caused by: java.lang.reflect.InvocationTargetException
MLT 	at jdk.internal.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
MLT 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
MLT 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
MLT 	at java.base/java.lang.Class.lookupCachedMethod(Class.java:3871)
MLT 	... 16 more
MLT Caused by: java.lang.IllegalArgumentException: Can not copy a non-root Method
MLT 	at java.base/java.lang.reflect.Method.copy(Method.java:158)
MLT 	... 20 more
MLT <<<
MLT 
```

For example, to rebuild the failed tests in <Jenkins URL>=https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder, use the following links:
03:25:24  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=11&JDK_IMPL=openj9&BUILD_LIST=system/mathLoadTest&PLATFORM=x86-64_linux&TARGET=MathLoadTest_all_special_12

**Comment by JasonFengJ9** [comment]
Another similar failure at an JDK8 internal build `Test_openjdk8_j9_special.system_x86-64_linux_jit_Nightly/600`:
```
MLT 02:50:02.002 - First failure detected by thread: load-1. Not creating dumps as no dump generation is requested for this load test
MLT 02:50:02.003 - Test failed
MLT   Failure num.  = 1
MLT   Test number   = 57
MLT   Test details  = 'JUnit[net.adoptopenjdk.test.bigdecimal.TestSuite057]'
MLT   Suite number  = 0
MLT   Thread number = 1
MLT >>> Captured test output >>>
MLT Test failed:
MLT java.lang.InternalError: java.lang.reflect.InvocationTargetException
MLT 	at java.lang.Class.newInternalError(Class.java:3795)
MLT 	at java.lang.Class.lookupCachedMethod(Class.java:3815)
MLT 	at java.lang.Class.getMethodHelper(Class.java:1263)
MLT 	at java.lang.Class.getMethod(Class.java:1214)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.getAnnotations(JUnit38ClassRunner.java:131)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.makeDescription(JUnit38ClassRunner.java:101)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.makeDescription(JUnit38ClassRunner.java:109)
MLT 	at org.junit.internal.runners.JUnit38ClassRunner.getDescription(JUnit38ClassRunner.java:95)
MLT 	at org.junit.runners.Suite.describeChild(Suite.java:123)
MLT 	at org.junit.runners.Suite.describeChild(Suite.java:27)
MLT 	at org.junit.runners.ParentRunner.getDescription(ParentRunner.java:352)
MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
MLT 	at net.adoptopenjdk.loadTest.adaptors.JUnitAdaptor.executeTest(JUnitAdaptor.java:130)
MLT 	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:182)
MLT 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
MLT 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
MLT 	at java.lang.Thread.run(Thread.java:823)
MLT Caused by: java.lang.reflect.InvocationTargetException
MLT 	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
MLT 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
MLT 	at java.lang.reflect.Method.invoke(Method.java:498)
MLT 	at java.lang.Class.lookupCachedMethod(Class.java:3812)
MLT 	... 16 more
MLT Caused by: java.lang.IllegalArgumentException: Can not copy a non-root Method
MLT 	at java.lang.reflect.Method.copy(Method.java:151)
MLT 	... 20 more
MLT <<<
```

**Comment by pshipton** [comment]
@mpirvu 
Perhaps related to https://github.com/eclipse/openj9/issues/11577

**Comment by dmitry-ten** [comment]
Today's grinder [run](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/15069/) passed. Will grind the specific variation that failed to check if the issue is still present. Note that the variation of the test corresponding to `Mode351` is now 11, instead of 12.

**Comment by dmitry-ten** [comment]
6 out of 50 [runs](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/15077/consoleText) with same error as the original crash, so the issue is still present. I did another 50 [runs](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/15089/) without JITServer and they all passed, so we can be pretty sure that it's a JITServer-specific problem

**Comment by dmitry-ten** [comment]
The latest nightly build fails with a slightly different exception:
```
12:50:52  MLT   Failure num.  = 1
12:50:52  MLT   Test number   = 12
12:50:52  MLT   Test details  = 'JUnit[net.adoptopenjdk.test.bigdecimal.TestSuite012]'
12:50:52  MLT   Suite number  = 0
12:50:52  MLT   Thread number = 0
12:50:52  MLT >>> Captured test output >>>
12:50:52  MLT Test failed:
12:50:52  MLT java.lang.ClassCastException: java.util.regex.Matcher incompatible with java.lang.reflect.Method
12:50:52  MLT 	at java.base/java.lang.Class.lookupCachedMethod(Class.java:4158)
12:50:52  MLT 	at java.base/java.lang.Class.getMethodHelper(Class.java:1444)
12:50:52  MLT 	at java.base/java.lang.Class.getMethod(Class.java:1395)
12:50:52  MLT 	at org.junit.internal.runners.JUnit38ClassRunner.getAnnotations(JUnit38ClassRunner.java:131)
12:50:52  MLT 	at org.junit.internal.runners.JUnit38ClassRunner.makeDescription(JUnit38ClassRunner.java:101)
12:50:52  MLT 	at org.junit.internal.runners.JUnit38ClassRunner.makeDescription(JUnit38ClassRunner.java:109)
12:50:52  MLT 	at org.junit.internal.runners.JUnit38ClassRunner.getDescription(JUnit38ClassRunner.java:95)
12:50:52  MLT 	at org.junit.runners.Suite.describeChild(Suite.java:123)
12:50:52  MLT 	at org.junit.runners.Suite.describeChild(Suite.java:27)
12:50:52  MLT 	at org.junit.runners.ParentRunner.getDescription(ParentRunner.java:352)
12:50:52  MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:136)
12:50:52  MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
12:50:52  MLT 	at net.adoptopenjdk.loadTest.adaptors.JUnitAdaptor.executeTest(JUnitAdaptor.java:130)
12:50:52  MLT 	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:182)
12:50:52  MLT 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
12:50:52  MLT 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
12:50:52  MLT 	at java.base/java.lang.Thread.run(Thread.java:866)
12:50:52  MLT <<<
```
and in another run it failed with:
```
[2021-06-24T17:50:16.907Z] MLT java.lang.IncompatibleClassChangeError
[2021-06-24T17:50:16.907Z] MLT 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2021-06-24T17:50:16.907Z] MLT 	at junit.framework.TestCase.runTest(TestCase.java:176)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestCase.runBare(TestCase.java:141)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestResult$1.protect(TestResult.java:122)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestResult.runProtected(TestResult.java:142)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestResult.run(TestResult.java:125)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestCase.run(TestCase.java:129)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestSuite.runTest(TestSuite.java:252)
[2021-06-24T17:50:16.908Z] MLT 	at junit.framework.TestSuite.run(TestSuite.java:247)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.Suite.runChild(Suite.java:128)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.Suite.runChild(Suite.java:27)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
[2021-06-24T17:50:16.908Z] MLT 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
[2021-06-24T17:50:16.908Z] MLT 	at net.adoptopenjdk.loadTest.adaptors.JUnitAdaptor.executeTest(JUnitAdaptor.java:130)
[2021-06-24T17:50:16.908Z] MLT 	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:182)
[2021-06-24T17:50:16.908Z] MLT 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2021-06-24T17:50:16.908Z] MLT 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2021-06-24T17:50:16.908Z] MLT 	at java.base/java.lang.Thread.run(Thread.java:866)

```

**Comment by dmitry-ten** [comment]
The current failures are for `MathLoadTest_all_special_5m_11`  which runs with options
```
 -Xgcpolicy:metronome -Xcompressedrefs  -XX:+UseJITServer
```
Notably, I grinded `MathLoadTest_all_special_5m_10` for 50 iterations and saw no failure. This test is identical to the previous one, but options are:
```
 -Xgcpolicy:metronome -Xnocompressedrefs  -XX:+UseJITServer
```
This seems to suggest that JITServer is not processing compressed references correctly in some rare cases.
We've had those problems in the past, but they were fixed. So far I haven't found any suspicious code related to compressed refs.
